storage: add test cases with zero-sized object IDs

jforissier opened the pull request:
The TEE Internal Core API specification (v1.3.1) explicitly allows the
use of an empty object ID in TEE_RenamePersistentObject(). The text is:

 newObjectID, newObjectIDLen: A buffer containing the new object
   identifier. The identifier contains arbitrary bytes, including the
   zero byte. The identifier length SHALL be less than or equal to
   TEE_OBJECT_ID_MAX_LEN and can be zero.

(note the mention: "and can be zero").

Therefore, add a couple of test cases in regression_6008. Both
(non-NULL, 0) and (NULL, 0) are tested for the parameters
(newObjectID, newObjectIDLen).

Since it follows from the above that zero-sized object IDs are indeed
valid, similar test cases are added to regression_6001
(TEE_CreatePersistentObject()) and regression_6002
(TEE_OpenPersistentObject()).

Signed-off-by: Jerome Forissier <jerome@forissier.org>

<!--
    If you are new to submitting pull requests to OP-TEE, then please have a
    look at the list below and tick them off before submitting the pull request.

    1. Read our contribution guidelines:
         https://optee.readthedocs.io/general/contribute.html

    2. Read the contribution section in Notice.md and pay extra attention to the
       "Developer Certificate of Origin" in the contribution guidelines.

    3. You should run checkpatch preferably before submitting the pull request.

    4. When everything has been reviewed, you will need to squash, rebase and
       add tags like `Reviewed-by`, `Acked-by`, `Tested-by` etc. More details
       about this can also be found on the link provided above.

    NOTE: This comment will not be shown in the pull request, so no harm keeping
    it, but feel free to remove it if you like.
-->

jforissier commented:
Depends on https://github.com/OP-TEE/optee_os/pull/5188.

etienne-lms reviewed (commented):
Should there be a failling test where oid is NULL and oid-length is non-zero? the test should gently refuse the object creation or renaming.

jforissier commented on host/xtest/regression_6000.c:
```diff
@@ -999,6 +1051,52 @@ static void xtest_tee_test_6008_single(ADBG_Case_t *c, uint32_t storage_id)
+
+	(void)ADBG_EXPECT_BUFFER(c, data_00, 10, out, count);
+
+	/* Object ID = (non-NULL, 0) */
```
Right, will fix

jforissier commented:
> Should there be a failling test where oid is NULL and oid-length is non-zero? the test should gently refuse the object creation or renaming.

Yes, maybe. In fact when adding such a test I noticed that the NULL pointer is not actually passed down to the TEE Internal API: there is a `TEE_Malloc()` in the storage TA so I need to fix this.

jforissier commented:
Update:
- Added commit "ta/storage: allow to pass a NULL objectId to open, create and rename" (I will move it before "storage: add test cases with zero-sized object IDs" when rebasing)
- Fixed comment

I didn't add any test with object ID (NULL, non-zero) because it is a bit difficult to do. The reason is the buffer is passed from the CA to the TA by `TEE_InvokeCommand()` in a `tmpref` parameter, and this invalid combination is rejected with `TEE_ERROR_INVALID_PARAMETER` so it doesn't even reach the TA. Some more code would be needed (such as a special command for the storage TA perhaps).

etienne-lms commented:
Right, it maybe not worth the pain (maintainance). There no such test neither on the data part.

jenswi-linaro commented:
`Acked-by: Jens Wiklander <jens.wiklander@linaro.org>`
